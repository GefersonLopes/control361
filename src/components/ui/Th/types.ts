export interface ThProps
  extends React.HTMLAttributes<HTMLTableHeaderCellElement> {
  children: React.ReactNode;
  className?: string;
}
