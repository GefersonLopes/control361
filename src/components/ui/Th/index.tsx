import clsx from "clsx";

import type { ThProps } from "./types";

function Th({ children, className }: ThProps) {
  return (
    <th
      className={clsx(
        "px-4 py-2 text-left font-medium tracking-wide",
        className,
      )}
    >
      {children}
    </th>
  );
}

export default Th;
