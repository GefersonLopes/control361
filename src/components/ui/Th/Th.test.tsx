import "@testing-library/jest-dom";

import { render, screen } from "@testing-library/react";

import Th from "../Th";

describe("Th component", () => {
  it("renders a th element with default classes and children", () => {
    render(
      <table>
        <thead>
          <tr>
            <Th>Header</Th>
          </tr>
        </thead>
      </table>,
    );
    const headerCell = screen.getByText("Header");
    expect(headerCell.tagName).toBe("TH");
    expect(headerCell).toHaveClass(
      "sticky",
      "top-0",
      "z-10",
      "bg-dark",
      "px-4",
      "h-[55px]",
      "text-center",
      "font-semibold",
      "font-poppins",
      "text-sm",
      "text-white",
      "tracking-wide",
    );
  });

  it("merges a custom className when provided", () => {
    render(
      <table>
        <thead>
          <tr>
            <Th className="extra-th custom">Col</Th>
          </tr>
        </thead>
      </table>,
    );
    const th = screen.getByText("Col");
    expect(th).toHaveClass("extra-th", "custom");
  });

  it("passes arbitrary HTML props to the th element", () => {
    render(
      <table>
        <thead>
          <tr>
            <Th data-testid="th-test" aria-label="Head" title="title-text">
              X
            </Th>
          </tr>
        </thead>
      </table>,
    );
    const th = screen.getByTestId("th-test");
    expect(th).toHaveAttribute("aria-label", "Head");
    expect(th).toHaveAttribute("title", "title-text");
  });

  it("matches the snapshot", () => {
    const { asFragment } = render(
      <table>
        <thead>
          <tr>
            <Th className="snap-class" data-extra="y">
              Snap
            </Th>
          </tr>
        </thead>
      </table>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
