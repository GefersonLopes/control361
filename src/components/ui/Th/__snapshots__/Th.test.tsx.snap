// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Th component matches the snapshot 1`] = `
<DocumentFragment>
  <table>
    <thead>
      <tr>
        <th
          class="sticky top-0 z-10 bg-dark px-4 h-[55px] text-center font-semibold  font-poppins text-sm text-white tracking-wide snap-class"
          data-extra="y"
        >
          Snap
        </th>
      </tr>
    </thead>
  </table>
</DocumentFragment>
`;
